Remotion/Data/DomainObjects/Queries/Configuration/QueryConfiguration.cs

```diff
@@ -51,6 +56,10 @@ public QueryConfiguration ()
       _properties.Add (_queryFilesProperty);
     }
 
+    public QueryConfiguration () : this(new AppContextProvider())
+    {
+    }
+
     public QueryConfiguration (params string[] configurationFiles) : this()
```

Comment:
We need an overload with IAppContextProvider

Reply:
Maybe I am misunderstanding you, but there is an overload with `IAppContextProvider`, defined in lines 43-57?

Reply:
We are missing the overload for the ctor that accepts configuration files. 

And while we're at it (and the pattern is a bit unfortunate here), but in theory, all overloads for ctors should converge to a single implementation unless there's a reason against it. Nominally, this change would be out of scope of the current task, unless the code needed to introduce IAppContextProvider creates more noise/less maintainable code than what we had before. Said differently, if the new ctor parameter requires we refactor the ctors first to have one implemention and several overloads, that's okay.

Reply:
Again, I'm hoping I got you right. I refactored the constructor(s) in b58204f27459d9d476d97917bc3e60a8f91dee96. IMO, it does not really cluster the change set or make noisy (given I interpreted you correctly).